2020/03/index.js

```diff
@@ -1,3 +1,53 @@
+const { promises: fs } = require('fs');
+
+async function getInput() {
+  return input = (await fs.readFile(`${__dirname}/input.txt`))
+    .toString()
+    .trim()
+    .split('\n')
+    .map(line => line.split(''));
+}
+
+async function p1(input, iIncrement = 3, jIncrement = 1) {
+  const tree = '#';
+  input = input
+    ? await Promise.resolve(input)
+    : await getInput();
+
+  let i = 0;
+  let numTrees = 0;
+  for(let j = 0; j < input.length; j += jIncrement) {
+    const row = input[j];
+    if(row[i % row.length] === tree) {
+      numTrees += 1;
+    }
+
+    i += iIncrement;
+  }
+
+  return numTrees;
+}
+
+async function p2() {
+  const input = await getInput();
+  const [a, b, c, d, e] = await Promise.all([
+    p1(input, 1, 1),
+    p1(input, 3, 1),
+    p1(input, 5, 1),
+    p1(input, 7, 1),
+    p1(input, 1, 2)
+  ]);
+
+  return a * b * c * d * e;
```

Comment:
Why this rather than a reduce chaining the promise.all?

Reply:
Yea that's probably better... Lol.